src/Service/TeamService.php

```diff
@@ -0,0 +1,212 @@
+
+        if ($returnCount) {
+            // find total count
+            return $teamRepository->findCountByCriteria($conditions);
+        }
+
+        // find data
+        return $teamRepository->findByCriteria($conditions, $this->requestParams['limit'], $this->requestParams['offset']);
+    }
+
+    /**
+     * @param Request $request
+     * @return Team|\Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function create(Request $request)
+    {
+        $this->getRequestParams($request);
+
+        /**
+         * @var $leagueRepository LeagueRepository
+         */
+        $leagueRepository = $this->em->getRepository(League::class);
+
+        // find league
+        $league = $leagueRepository->find($this->requestParams['league_id']);
+        if (empty($league)) {
+            $this->error = ApiCodes::ERR_CODE_LEAGUE_NOT_FOUND;
+            return $this->responseJson([], Response::HTTP_BAD_REQUEST);
+        }
+
+        // create team
+        $team = new Team();
+        $team->setName($this->requestParams['name']);
+        $team->setStrip($this->requestParams['strip']);
+        $team->setLeague($league);
+
+        try {
+            $this->em->persist($team);
+            $this->em->flush();
+        } catch (UniqueConstraintViolationException $e) {
+            $this->error = ApiCodes::ERR_CODE_TEAM_DUPLICATE_NAME;
+            return $this->responseJson([], Response::HTTP_BAD_REQUEST);
+        }
+
+        return $team;
+    }
+
+    /**
+     * @param $id
+     * @param Request $request
+     * @return Team|null|\Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function update($id, Request $request)
+    {
+        $this->getRequestParams($request);
+
+        /**
+         * @var $teamRepository TeamRepository
+         */
+        $teamRepository = $this->em->getRepository(Team::class);
+
+        $teamEntity = $teamRepository->find($id);
+
+        if (empty($teamEntity)) {
+            $this->error = ApiCodes::ERR_CODE_TEAM_NOT_FOUND;
+            return $this->responseJson([], Response::HTTP_BAD_REQUEST);
+        }
+
+        $changed = false;
+        if (!empty($this->requestParams['name'])) {
+            $teamEntity->setName($this->requestParams['name']);
+            $changed = true;
+        }
+        if (!empty($this->requestParams['strip'])) {
+            $teamEntity->setStrip($this->requestParams['strip']);
+            $changed = true;
+        }
+        if (!empty($this->requestParams['league_id'])) {
+            /**
+             * @var $leagueRepository LeagueRepository
+             */
+            $leagueRepository = $this->em->getRepository(League::class);
+
+            // find league
+            $league = $leagueRepository->find($this->requestParams['league_id']);
+            if (empty($league)) {
+                $this->error = ApiCodes::ERR_CODE_LEAGUE_NOT_FOUND;
+                return $this->responseJson([], Response::HTTP_BAD_REQUEST);
```

Comment:
Violation of the [SoC](https://en.wikipedia.org/wiki/Separation_of_concerns) principle. Services have no responsibility to return a json respons.